src/nodes/NodeManager.ts

```diff
@@ -410,6 +459,7 @@ class NodeManager {
     force: boolean = false,
     timeout?: number,
     tran?: DBTransaction,
+    options: { signal?: AbortSignal } = {},
```

Comment:
The `timeout` and the `signal` should be replaced with a `ContextTimed` object.

Use the `timedCancellable` decorator here.

This is a bit more complicated than original design.

The reason is that the `setNode` can schedule background tasks or not.

In the case of `block: false`, then everything runs within the same promise lifecycle. In which case the `timer` and the `signal` gets propagated down to all the downstream functions. Now this would mean:

```
public setNode(
  nodeId: NodeId,
  nodeAddress: NodeAddress,
  block?: boolean,
  force?: boolean,
  ctx?: Partial<ContextTimed>
): PromiseCancellable<void>;
@ready(new nodesErrors.ErrorNodeManagerNotRunning())
@timedCancellable(false, 1000)
public async setNode(
  nodeId: NodeId,
  nodeAddress: NodeAddress,
  block: boolean = true,
  force: boolean = false,
  @context ctx: ContextTimed
): Promise<void> {

  // ...

}
```

Reply:
Once this change takes place. The meaning of `timeout` is different. The original `timeout` corresponds to the connection establishment timeout that occurs for every proxy connection.

However in this case, that's not what what the timer means. Instead timer is now defaulted to 1000 ms. Which is the total amount of time allowed to complete a `setNode` operation.

Reply:
The `ctx.timer.getTimeout()` can then be used to pass into `pingNode` which has independent timeouts that is all occurring in parallel. This works for the `block: true`.

What about `block: false`, in this case, we would propagate the `ctx.timer.delay` into the `TaskDeadline`, which itself is then propagated to the `pingNode` running inside the task handler function.

Reply:
This applies to the intermediate `garbageCollectOldNode` too.

In the new garbage collection mechanism where we are maintaining pending nodes. We would be iterating the existing nodes in a bucket in parallel and pinging them. However we may wish still abide by the k-parallel constant.

This `k parallel constant` should be written to the class as a parameter.

Reply:
The `timedCancellable` conversions are done fr this PR. `setNode` was updated to be a `timedCancellable` but the timer is for the overall task completeion. The `pingNode`s as part of the garbage collection are started with their own timers set by the `pingTimeout` parameter.

Further `timedCancellable` conversion will be done in a later PR.